pages/docs/agent-kit/ai-agent-network-state-routing.mdx

```diff
@@ -0,0 +1,280 @@
+    - Re-calls the router with the new state, which either quits or runs another agent
+
+<hr />
+
+## Network state
+
+<Callout variant="info">
+  Network state passes context between different Agents in a series of calls.  It stores ouputs from prior calls, and has a KV store for additional data from tools or custom code.
+</Callout>
+
+Network state is the memory of your agent workflow.  It stores two things:
+
+1. **History Management**: It keeps track of all agent interactions, including prompts, responses, and tool calls
+2. **Key-Value Storage**: It provides a simple key-value store for sharing data and facts between different agent calls
+
+Without this state, each agent call would be isolated and unable to build upon previous interactions to solve complex tasks.  The history and key value stores are used automatically by the network to give each agent context about what happened before.  We’ll dive into both.
+
+### History
+
+The history system maintains a chronological record of all agent interactions in your network. Each interaction is stored as an `InferenceResult`, which includes:
+
+- `agent`: The agent that created this result
+- `input`: The original input
+- `system`: System instructions
+- `prompt`: The complete prompt
+- `output`: Agent output
+- `toolCalls`: Tool calls and their results
+- `raw`: Raw API responses
+
+### Using state in agent prompts
+
+The network state keeps track of every agent interaction, building a chronological list of messages in memory.  When the network runs, it calls the `onStart` lifecycle hook with the given agent, network, user input, system, and history from memory:
+
+```jsx
+const agent = new Agent({
+  name: "Code writer",
+  lifecycles: {
+    onStart: ({
+      agent,
+      network, // has the entire state
+      input,
+
+      system, // The system prompt for the agent
+      history, // An array of messages
+    }) => {
+      // Return the system prompt (the first message), and any history added to the
+      // model's conversation.
+      return { system, history };
+    }
+  },
+});
+```
+
+This lifecycle hook can be used to modify the system prompt and history used within each agent.
+
+### The state key-value store
+
+The KV store is a simple way to store information between agent calls.  Think of it like a shared memory that all your agents and tools can access.  Here's how to use it:
+
+```tsx
+// Store a value
+network.state.kv.set("user-name", "Alice");
+
+// Get a value
+const name = network.state.kv.get<string>("user-name");
+
+// Delete a value
+network.state.kv.delete("user-name");
+
+// Check if a value exists
+const exists = network.state.kv.has("user-name");
+```
+
+Common uses for the KV store include:
+
+- Storing intermediate results that other agents might need within lifecycles
+- Storing user preferences or context
+- Passing data between tools and agents
+
+Here’s a tool which stores files created by a coding agent the network’s key-value state:
```

Comment:
Maybe callout that this is using Zod - most people probably know when they see `z`, but if they don't it's confusing